tensorflow/contrib/makefile/rename_protobuf.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+# Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+# This script modifies the downloaded protobuf library and the TensorFlow source
+# to put all protobuf-related symbols into the google::protobuf3 namespace
+# instead of the default google::protobuf. This is necessary to work around
+# linking issues for applications that use protobuf v2 already, and want to
+# adopt TensorFlow, since otherwise the two libraries have duplicate function
+# symbols that clash. It also renames all the include paths to google/protobuf3
+# throughout the protobuf and TensorFlow code.
+# This is a massive hack, and if possible it's recommended that you switch your
+# whole application to protobuf v3 so there's no mismatch with TensorFlow.
```

Comment:
(Is it worth mentioning that, although you've gotten something to work, there's no guarantee that this solution is bulletproof?)